File: contest/ICPC/2014/欧洲/NWERC西北欧/all/centsavings/submissions/accepted/tobi-ID.cpp
#include <iostream>
#include <algorithm>
#include <climits>

using namespace std;

#define FOR(i,a,b) for (int i = (a); i < (b); i++)
#define FORD(i,a,b) for (int i = int(b)-1; i >= (a); i--)
#define ROUND(x) ((((x)+5)/10)*10)

int I, D, prices[2005], dp[21], mini;

int main() {
	cin >> I >> D;
	FOR(i, 0, I) cin >> prices[i];
	dp[0] = 0; // cost is zero with 0 divider
	FOR(i, 1, D+1) dp[i] = INT_MAX/2; // you have to place less dividers first
	FOR(i, 0, I) {
		FORD(j, 1, D+1) dp[j] = min(dp[j] + prices[i], ROUND(dp[j - 1] + prices[i])); // either just add item or place divider after item
		dp[0] += prices[i]; // no divider, no choice
	}
	mini = INT_MAX/2;
	FOR(i, 0, D+1) mini = min(mini, dp[i]);
	cout << ROUND(mini) << endl;
	return 0;
}

File: contest/ICPC/2014/欧洲/NWERC西北欧/all/centsavings/submissions/accepted/tobiID.java
import java.io.*;
import java.util.*;

public class tobiID {
	static int I, D;
       	static int prices[] = new int[2005];
	static int dp[] = new int[21];
	static int pSum[] = new int[2005];

	public static int ROUND(int x) {
		return ((((x)+5)/10)*10);
	}

	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		I = sc.nextInt();
		D = sc.nextInt();
		for (int i = 0; i < I; i++) prices[i] = sc.nextInt();
		dp[0] = 0; // cost is zero with 0 divider
		for (int i = 1; i < D+1; i++) dp[i] = Integer.MAX_VALUE/2;
		for (int i = 0; i < I; i++) {
			for (int j = D; j >= 1; j--) dp[j] = Math.min(dp[j] + prices[i], ROUND(dp[j - 1] + prices[i])); // either just add item or place divider after item
			dp[0] += prices[i]; // no divider, no choice
		}
		int mini = Integer.MAX_VALUE/2;
		for (int i = 0; i < D+1; i++) mini = Math.min(mini, dp[i]);
		System.out.println(ROUND(mini));
	}
}
